tests/test_clustering.py

```diff
@@ -0,0 +1,61 @@
+from numpy import array, ndarray
+
+from app.clustering import CDBSCAN, CAgglomerativeClustering, CKMeans
+
+
+def test_ckmeans():
+    """Test CKMeans clustering."""
+    data = array([[1, 2], [1, 4], [1, 0], [10, 2], [10, 4], [10, 0]])
+    kmeans = CKMeans(
+        n_clusters=2,
+        init="k-means++",
+        n_init="auto",
+        max_iter=300,
+        tol=1e-4,
+        verbose=0,
+        random_state=42,
+        copy_x=True,
+        algorithm="lloyd",
+    )
+    labels = kmeans.cluster(data)
+    assert isinstance(labels, ndarray), "Output labels should be an ndarray."
+    assert len(labels) == len(data), "Each data point should have a label."
```

Comment:
you do not test the functionality of the custom clustering - this test is an integration test. Please also write unit tests that check that the sci-kit learn method was called with given parameters, you mock for scikit_learn model - do not call it in unit tests